src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -331,6 +332,9 @@ def save_as_table(
             table_exists: Optional parameter to specify if the table is known to exist or not.
                 Set to ``True`` if table exists, ``False`` if it doesn't, or ``None`` (default) for automatic detection.
                 Primarily useful for "append" and "truncate" modes to avoid running query for automatic detection.
+            override_condition: Specifies the override condition to perform atomic targeted delete-insert.
+                Can only be used when ``mode`` is "append" and the table exists. Rows matching the
```

Comment:
my intuition is this shall be put under "overwrite" mode -- it's an advanced feature for selective "overwrite".